core/src/main/java/org/apache/carbondata/core/datastore/block/TableBlockInfo.java

```diff
@@ -105,6 +110,14 @@ public int compare(Distributable o1, Distributable o2) {
   public TableBlockInfo(String filePath, long blockOffset, String segmentId,
       String[] locations, long blockLength, ColumnarFormatVersion version,
       String[] deletedDeltaFilePath) {
+    this(null, filePath, blockOffset, segmentId,
```

Comment:
Need to remove this constructor ? Because if somebody don't set table id. Cache logic may go wrong ?